static/js/pm_list.js

```diff
@@ -59,52 +78,157 @@ export function update_private_messages() {
-    }
 }
 
 export function update_dom_with_unread_counts(counts) {
     update_private_messages();
     set_count(counts.private_message_count);
 }
 
-function should_expand_pm_list(filter) {
-    const op_is = filter.operands("is");
+function update_collapsed_private_messages_section() {
+    // when we need to update the collapsed private messages section, we first check
+    // for any active conversation thread which is supposed to be present outside to it.
+    const conversations = pm_list_data.get_conversations();
+    const active_conversation = conversations.find((conversation) => conversation.is_active);
+
+    if (active_conversation) {
+        // If we find an active conversation thread satisfying the requirement, then we update our
+        // DOM with it's node.
+        const node = [pm_list_dom.keyed_pm_li(active_conversation)];
+        const new_dom = pm_list_dom.pm_ul(node);
+        const $container = ui.get_content_element($("#sidebar-private-messages-container"));
+        function replace_content(html) {
+            $container.html(html);
+        }
 
-    if (op_is.length >= 1 && op_is.includes("private")) {
-        return true;
+        function find() {
+            return $container.find("ul");
+        }
+        vdom.update(replace_content, find, new_dom, prior_dom);
+        prior_dom = new_dom;
+    } else {
+        // else mark the section as collapsed.
+        close();
     }
+}
 
-    const op_pm = filter.operands("pm-with");
+export function show_all_private_messages_view() {
+    $("#private_messages_list").addClass("active_private_messages_list");
+    $("#more-private-conversations-sidebar-title").addClass(
+        "active-more-private-conversations-sidebar-title",
+    );
+}
 
-    if (op_pm.length !== 1) {
-        return false;
-    }
+function hide_all_private_messages_view() {
+    $("#private_messages_list").removeClass("active_private_messages_list");
+    $("#more-private-conversations-sidebar-title").removeClass(
+        "active-more-private-conversations-sidebar-title",
+    );
+}
 
-    const emails_strings = op_pm[0];
-    const emails = emails_strings.split(",");
+export function handle_narrow_activated(filter) {
+    const active_filter = filter;
+    const is_all_private_message_view = _.isEqual(active_filter.sorted_term_types(), [
+        "is-private",
+    ]);
+    const is_private_message_narrow = active_filter.operands("pm-with").length !== 0;
+
+    if (!is_all_private_message_view) {
+        hide_all_private_messages_view();
+    }
+    if (is_private_message_narrow && private_messages_collapsed) {
+        update_collapsed_private_messages_section();
+    }
+    if (!is_private_message_narrow && !is_all_private_message_view) {
+        deselect_private_messages_section();
+    }
+}
 
-    const has_valid_emails = people.is_valid_bulk_emails_for_compose(emails);
+export function handle_narrow_deactivated() {
+    if (private_messages_collapsed) {
+        close();
+    }
```

Comment:
I don't think it makes sense to call `close` again if we're already closed; it's very confusing. Calling `update_private_messages` to potentially adjust the visible-but-collapsed row seems right.

Reply:
Agreed; thanks!